app/src/main/java/com/checkin/app/checkin/Home/NearbyRestaurantAdapter.kt

```diff
@@ -1,43 +1,155 @@
+import androidx.viewpager.widget.PagerAdapter
+import androidx.viewpager.widget.ViewPager
 import com.checkin.app.checkin.Home.model.NearbyRestaurantModel
 import com.checkin.app.checkin.R
 import com.checkin.app.checkin.Utility.BaseViewHolder
+import com.rd.PageIndicatorView
+import com.rd.animation.type.AnimationType
+import java.net.URL
+import java.util.*
+import android.graphics.BitmapFactory
+import android.graphics.Bitmap
+import android.nfc.Tag
+import android.widget.TextView
+import butterknife.BindView
+import com.bumptech.glide.Glide
+
+
+class NearbyRestaurantAdapter(var context: Context) : RecyclerView.Adapter<BaseViewHolder<Any>>() {
+
+
 
-class NearbyRestaurantAdapter : RecyclerView.Adapter<BaseViewHolder<Any>>() {
     var data: List<NearbyRestaurantModel> = emptyList()
 
     fun updateData(newData: List<NearbyRestaurantModel>) {
+
         data = newData
         notifyDataSetChanged()
     }
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder<Any> {
         val view = LayoutInflater.from(parent.context).inflate(viewType, parent, false)
-        return if (viewType == R.layout.item_home_restaurant_banner) NearbyRestaurantViewHolder(view) as BaseViewHolder<Any> else AdvertisementViewHolder(view)
+        return if (viewType == R.layout.item_home_restaurant_banner) NearbyRestaurantViewHolder(view)
+                as BaseViewHolder<Any> else AdvertisementViewHolder(view)
     }
 
-    override fun getItemCount(): Int = data.count() + 1
+    override fun getItemCount(): Int = data.count()
 
-    override fun getItemViewType(position: Int): Int = if (position == 2) R.layout.item_advertisement_banner else R.layout.item_home_restaurant_banner
+    override fun getItemViewType(position: Int): Int = if (position == 2)
+        R.layout.item_advertisement_banner else R.layout.item_home_restaurant_banner
 
     override fun onBindViewHolder(holder: BaseViewHolder<Any>, position: Int) {
         holder.bindData(data[position])
     }
 
     inner class NearbyRestaurantViewHolder(itemView: View) : BaseViewHolder<NearbyRestaurantModel>(itemView) {
+
+
+        var imageView:ImageView=itemView.findViewById(R.id.iv_home_restaurant_banner_main)
+
+        var image_title:TextView=itemView.findViewById(R.id.tv_home_restaurant_banner_image_title)
+
+        var ratings:TextView=itemView.findViewById(R.id.tv_home_restaurant_banner_ratings)
+        var resturantName:TextView=itemView.findViewById(R.id.tv_home_restaurant_banner_restaurant_name)
+
+        init {
+
+
+            imageView.scaleType = ImageView.ScaleType.FIT_XY
+        }
+
         override fun bindData(data: NearbyRestaurantModel) {
-            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+
+            image_title.setText(data.imageTitle)
+            ratings.setText(data.ratings)
+            resturantName.setText(data.restaurantName)
+            Glide.with(context).asBitmap()
```

Comment:
That's not how we use Glide. First of all, you're supposed to use GlideApp (mentioned in docs).
Anyways, we have a utility function wrapper over Glide used everyone in code, `Utils.loadImageOrDefault`
Search its usage in code if you don't understand how to use it